src/main/java/cz/mzk/fedora/solrcorrection/SolrClientApi.java

```diff
@@ -0,0 +1,64 @@
+package cz.mzk.fedora.solrcorrection;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.function.Consumer;
+
+public class SolrClientApi {
+
+    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    public static SolrClient createSolrClient(String solrHost) {
+        return new HttpSolrClient.Builder(solrHost)
+                .withConnectionTimeout(10000)
+                .withSocketTimeout(60000)
+                .build();
+    }
+
+    public static void singleRequestAndApply(SolrQuery solrQuery, SolrClient solrClient,
+                                             Consumer<SolrDocument> consumer)
+            throws SolrServerException, IOException {
+        long numFound = queryForNumFound(solrQuery, solrClient);
+
+        if(numFound != 0) {
+            solrQuery.setRows(Math.toIntExact(numFound));
+            SolrDocumentList docs = queryForSolrDocList(solrQuery, solrClient);
+            forEachDoc(docs, consumer);
+        } else {
+            System.out.println("NumFound is 0!");
+        }
+    }
+
+    public static void forEachDoc(SolrDocumentList docs, Consumer<SolrDocument> consumer) {
```

Comment:
neni potreba, kdyz tak muzes vyuzit Stream API a aplikovat consumera
napr https://medium.com/swlh/understanding-java-8s-consumer-supplier-predicate-and-function-c1889b9423d